src/components/seo.js

```diff
@@ -77,7 +81,7 @@ const SEO = ({ canonical, description,image, schemaMarkup, title,children }) =>
           />
           }`}
       </noscript>
-      {canonical &&   <link rel="canonical" href={canonical} />}
+      {canonicalUrl &&   <link rel="canonical" href={canonicalUrl} />}
```

Comment:
Well, based on the change at line 38-41, it seems `canonicalUrl would never be `null/false` so we can remove this conditional from here and use it directly